File: src/kyu_7/digits_explosion/python/digits_explosion_test.py
import codewars_test as test
from solution import explode

sample_test_cases = [
    # (input, expected),
    ('312', '333122'),
    ('102269','12222666666999999999'),
    ('0', ''),
    ('000', ''),
    ('123', '122333'),
]

@test.describe('Sample tests')
def sample_tests():
    for s, expected in sample_test_cases:
        @test.it(f'explode({s!r})')
        def _():
            test.assert_equals(explode(s), expected)

File: src/kyu_7/digits_explosion/c/digits_explosion_test.c
#include <criterion/criterion.h>

extern void do_test (const char *digits, const char *expected);

Test(tests_suite, sample_tests)
{
  do_test("", "");
  do_test("0", "");
  do_test("1", "1");
  do_test("2", "22");
  do_test("3", "333");
  do_test("4", "4444");
  do_test("5", "55555");
  do_test("6", "666666");
  do_test("7", "7777777");
  do_test("8", "88888888");
  do_test("9", "999999999");
  do_test("000", "");
  do_test("312", "333122");
  do_test("102269","12222666666999999999");
}